tests/e2e/steps_permissionless_ics.go

```diff
@@ -157,8 +166,28 @@ func stepsPermissionlessICS() []Step {
 					},
 					ChainID("provi"): e2e.ChainState{
 						HasToValidate: &map[ValidatorID][]ChainID{
-							ValidatorID("alice"): {"consu"},
-							ValidatorID("bob"):   {"consu"}, // bob is still a validator on consu chain
+							ValidatorID("alice"): {"cons1"},
+							ValidatorID("bob"):   {"cons1"}, // bob is still a validator on consu chain
+							ValidatorID("carol"): {},
+						},
+					},
+				},
+			},
+		},
+		// Remove permissionless chain
+		[]Step{
+			{
+				Action: RemoveConsumerChainAction{
+					Chain:         ChainID("provi"),
+					From:          ValidatorID("alice"),
+					ConsumerChain: ChainID("cons1"),
+				},
+				State: State{
+					ChainID("provi"): e2e.ChainState{
+						ConsumerChains: &map[ChainID]bool{"cons2": true}, // Consumer chain "consu1" is now removed
+						HasToValidate: &map[ValidatorID][]ChainID{
+							ValidatorID("alice"): {"cons1"},
+							ValidatorID("bob"):   {"cons1"}, // bob is still a validator on consu chain
```

Comment:
```suggestion
							ValidatorID("bob"):   {"cons1"}, // bob is still a validator on cons1 chain
```
?

Reply:
Actually, `cons1` is removed but validators still have to validate according to this query. This seems wrong: Why would you need to validate a chain that has been stopped? I'll publish a fix on this in a bit.

Reply:
this is based on the description from the ADR that the final delete message will be called after unbonding period has passed.

**Stopped phase.** 
Lastly, the owner of a chain can choose to exit ICS by executing a `MsgRemoveConsumer`.
After some period of time (e.g., provider's unbonding period), all state related to the stopped consumer chain can be removed.

As discussed , we can open an issue and change the behaviour to what HasToValidate() should return, e.g. only report validators of active chains.